src/MinimalDelegation.sol

```diff
@@ -47,6 +47,12 @@ contract MinimalDelegation is IERC7821, IKeyManagement, ERC1271, EIP712, ERC4337
 
     /// @dev The mode is passed to allow other modes to specify different types of opData decoding.
     function _authorizeOpData(bytes32, Call[] calldata calls, bytes calldata opData) private view {
+        if (msg.sender == ENTRY_POINT()) {
```

Comment:
I wonder if it would at all be worth it to instead have a specific mode dedicated to userOp handling? 

also with this piece of code we should just make sure that reentrancy on the entrypoint contract can't somehow call back into this function and execute whatever

Reply:
hmm, yeah could potentially explore because opData will be encoded differently

Reply:
Actually I don't think we should have a separate mode for entrypoint:
- I see the decision of wrapping a call in a userOp or not as being made strictly after the calls are encoded
- Thus by adding a specific mode we require the dapp (or wallet using 5792) to know whether or not it wants to do gas sponsorship when creating the calldata, I worry this will make the experience cumbersome